test/e2e/build_test.go

```diff
@@ -41,6 +41,35 @@ var _ = Describe("Podman build", func() {
 		Expect(session).Should(ExitCleanly())
 	})
 
+	It("podman image prune should clean build cache", func() {
+		// try writing something to persistent cache
+		session := podmanTest.Podman([]string{"build", "-f", "build/buildkit-mount/Containerfilecachewrite"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(ExitCleanly())
+
+		// try reading something from persistent cache
+		session = podmanTest.Podman([]string{"build", "-f", "build/buildkit-mount/Containerfilecacheread"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(ExitCleanly())
+		Expect(session.OutputToString()).To(ContainSubstring("hello"))
+
+		// Prune build cache
+		session = podmanTest.Podman([]string{"image", "prune", "-f", "--build-cache"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(ExitCleanly())
+
+		// try reading something from persistent cache should fail
+		session = podmanTest.Podman([]string{"build", "-f", "build/buildkit-mount/Containerfilecacheread"})
+		session.WaitWithDefaultTimeout()
+		if IsRemote() {
+			Expect(session.OutputToString()).To(ContainSubstring("open '/test/world': No such file or directory"))
+			Expect(session).Should(ExitWithError(1, "cat /test/world\": while running runtime: exit status 1"))
+		} else {
+			Expect(session.ErrorToString()).To(ContainSubstring("open '/test/world': No such file or directory"))
+			Expect(session).Should(ExitWithError(1, "open '/test/world': No such file or directory"))
+		}
```

Comment:
This really needs a comment why this is different. is this even expected or just a straight up remote build bug that the std streams are not preserved correctly?

The second check is still duplicating the same message ceck twice so you should really drop the `ErrorToString()` as it is a dup. And please consider splitting the same message into a var and not duplicate it each time.

Reply:
> This really needs a comment why this is different. is this even expected or just a straight up remote build bug that the std streams are not preserved correctly?

This is a bug which needs to fixed on `remote build` end it seems `remote build` throws some messages on `stdout` instead of `stderr`. I'll try investigating and see if its an easy fix.